carta/image.py

```diff
@@ -42,25 +40,25 @@ def __init__(self, session, image_id, file_name):
         self._frame = Macro("", self._base_path)
 
     @classmethod
-    def new(cls, session, path, hdu, append, complex, expression, make_active=True, update_directory=False):
+    def new(cls, session, directory, file_name, hdu, append, image_arithmetic, make_active=True, update_directory=False):
         """Open or append a new image in the session and return an image object associated with it.
 
-        This method should not be used directly. It is wrapped by :obj:`carta.session.Session.open_image` and :obj:`carta.session.Session.append_image`.
+        This method should not be used directly. It is wrapped by :obj:`carta.session.Session.open_image`, :obj:`carta.session.Session.open_complex_image` and :obj:`carta.session.Session.open_LEL_image`.
 
         Parameters
         ----------
         session : :obj:`carta.session.Session`
             The session object.
-        path : string
-            The path to the image file, either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory.
+        directory : string
+            The directory containing the image file, or the base directory for the LEL arithmetic expression, as an absolute path relative to the CARTA backend's root directory.
+        file_name : string
+            The name of the image file, or a LEL arithmetic expression.
         hdu : string
             The HDU to open.
         append : boolean
             Whether the image should be appended.
-        complex : boolean
-            Whether the image is complex.
-        expression : a member of :obj:`carta.constants.ArithmeticExpression`
-            Arithmetic expression to use if opening a complex-valued image. Ignored if the image is not complex.
+        image_arithmetic : boolean
+            Whether the file name should be interpreted as a LEL expression. Ignored if the image is not complex.
```

Comment:
A complex expression is just a specific kind of LEL arithmetic expression. We have an additional session function to handle this special case, but the `new` function doesn't distinguish between them. `open_complex_image` constructs a LEL expression string from the complex expression constant and the file name.  There is no need to mention complex images here specifically.

I would actually like to rename the `ComplexExpression` enum, because it's not an expression (and the parameter with this enum value should not have the same name as the parameters with the string expressions). @kswang1029 would it make sense to call this a "complex part" or a "complex coordinate"?

Reply:
OK, I have remove complex image in the description in b5bffc5.